pyat/at/physics/linear.py

```diff
@@ -525,49 +527,86 @@ def get_tune(ring, method='linopt', **kwargs):
         fmin/fmax: determine the boundaries within which the tune is
         located [default 0->1]
         hann: flag to turn on hanning window [default-> False]
-
+        remove_dc: Removes the mean offset of oscillation data
 
     OUTPUT
         tunes = np.array([Qx,Qy])
     """
 
-    def gen_centroid(ring, ampl, nturns):
-        p0 = numpy.zeros((6, 2))
-        p0[0, 0] = ampl
-        p0[2, 1] = ampl
+    def gen_centroid(ring, ampl, nturns, dp, remove_dc):
+        orbit, _ = find_orbit4(ring, dp)
+        ld, _, _, _ = linopt(ring, dp)
```

Comment:
To avoid a double computation of `find_orbit4`, you should use
```
orbit, _ = find_orbit4(ring, dp)
ld, _, _, _ = linopt(ring, dp, orbit=orbit)
```